src/jltypes.c

```diff
@@ -1154,6 +1166,7 @@ static void cache_insert_type_set(jl_datatype_t *val, uint_t hv)
 
 jl_svec_t *cache_rehash_set(jl_svec_t *a, size_t newsz)
 {
+    newsz = newsz ? bit_ceil(newsz) : 0;
```

Comment:
This function previously had an un-documented requirement that `newsz` be a power-of-two, which gave @gbaraldi some trouble recently.